kie-gwthelper-maven-plugin/src/main/java/org/kie/maven/gwthelper/plugin/AddSourceMojo.java

```diff
@@ -0,0 +1,191 @@
+        if (isValidGwtModule(source)) {
+            File sources = new File(source.getAbsolutePath() + SRC_MAIN_JAVA);
+            checkReadableDirectory(sources);
+            this.project.addCompileSourceRoot(sources.getAbsolutePath());
+            if (getLog().isInfoEnabled()) {
+                getLog().info("Source directory: " + sources + " added.");
+            }
+            File resources = new File(source.getAbsolutePath() + SRC_MAIN_RESOURCES);
+            this.project.addCompileSourceRoot(resources.getAbsolutePath());
+            if (getLog().isInfoEnabled()) {
+                getLog().info("Source directory: " + resources + " added.");
+            }
+        }
+    }
+
+    /**
+     * Method to check if in the given directory is a <i>valid</i> <b>Gwt</b> module, i.e. it contains a "src/main/resources/./.gwt.xml" file
+     * eventually matching the <b>includes/excludes</b> patterns
+     * @param toCheck
+     * @return
+     * @throws MojoExecutionException
+     */
+    private boolean isValidGwtModule(File toCheck) throws MojoExecutionException {
+        getLog().debug("isValidGwtModule " + toCheck.getAbsolutePath());
+        File resources = new File(toCheck.getAbsolutePath() + SRC_MAIN_RESOURCES);
+        if (!resources.exists()) {
+            return false;
+        }
+        boolean toReturn = false;
+        try {
+            final Optional<Path> first = Files.walk(Paths.get(resources.getAbsolutePath()))
+                    .filter(Files::isRegularFile)
+                    .filter(path -> path.toString().endsWith(".gwt.xml")).findFirst();
+            if (first.isPresent()) {
+                String fileName = first.get().getFileName().toString();
+                if (StringUtils.isNotEmpty(includes)) {
+                    toReturn = matchPattern(fileName, includes);
+                } else if (StringUtils.isNotEmpty(excludes)) {
+                    toReturn = !matchPattern(fileName, excludes);
+                } else {
+                    toReturn = true;
+                }
+            }
+        } catch (IOException e) {
+            String errorMessage = StringUtils.isEmpty(e.getMessage()) ? e.getClass().getName() : e.getMessage();
+            errorMessage += " while analyzing " + toCheck.getAbsolutePath();
+            throw new MojoExecutionException(errorMessage);
+        }
+        return toReturn;
+    }
+
+    /**
+     * Method to check if the given directory is a <b>Maven</b> module (it contains <b>pom.xml</b>)
+     * @param toCheck
+     * @return
+     */
+    private boolean isMavenModule(File toCheck) {
+        return toCheck.isDirectory() && toCheck.list() != null && Arrays.asList(toCheck.list()).contains("pom.xml");
+    }
+
+    /**
+     * Method to check if the given String contains one of the comma-separated pattern.
+     * Matching is done with String.contains()
+     * @param toCheck
+     * @param pattern
+     * @return
```

Comment:
I think Javadoc complains about empty return tags. I think it's OK to leave out the return tag if, the result is already described in method description.

Reply:
_Add descriptions_